pkg/e2e/operators/cloudingress/deletetapps2.go

```
@@ -0,0 +1,146 @@
+			returnAnnotations(h, ingressControllerName)
+			log.Print("Added the Annotations back to IngressOperator")
+			removeIngressController(h, ingressControllerName)
+			log.Print("Called removeIngressController ")
+			ingressControllerExists(h, ingressControllerName, false)
+			log.Print("ingressController Exist shouldn't throw any errors'")
+		})
+	})
+})
+
+func removeIngressController(h *helper.H, name string) {
+	_, exists, index := appIngressExits(h, false, name)
+	// only remove the secondary ingress if it already exist in the publishing strategy
+	if exists {
+		removeAppIngress(h, index)
+		//wait 2 minute for all resources to be deleted
+		time.Sleep(time.Duration(120) * time.Second)
+	}
+}
+func removeAnnotation(h *helper.H, name string) operatorv1.IngressController {
+	var ingressController operatorv1.IngressController
+	log.Print("Gonna start the annotation deletion process")
+	ingresscontroller, err := h.Dynamic().Resource(schema.GroupVersionResource{Group: "operator.openshift.io", Version: "v1", Resource: "ingresscontrollers"}).Namespace("openshift-ingress-operator").Get(context.TODO(), name, metav1.GetOptions{})
+	log.Print("Getting the ingressController object")
+	Expect(err).NotTo(HaveOccurred())
+
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(ingresscontroller.Object, &ingressController)
+	Expect(err).NotTo(HaveOccurred())
+	temp := ingressController.ObjectMeta
+	if temp.Annotations["Owner"] == "cloud-ingress-operator" {
+		delete(temp.Annotations, "Owner")
+	}
+	ingressController.ObjectMeta = temp
+	ingresscontroller.Object, err = runtime.DefaultUnstructuredConverter.ToUnstructured(&ingressController)
+	Expect(err).NotTo(HaveOccurred())
+
+	ingresscontroller, err = h.Dynamic().Resource(schema.GroupVersionResource{Group: "operator.openshift.io", Version: "v1", Resource: "ingresscontrollers"}).Namespace("openshift-ingress-operator").Update(context.TODO(), ingresscontroller, metav1.UpdateOptions{})
+	return ingressController
```

since the err is not returned, I would add a `Expect(err).NotTo(HaveOccurred())` prior to the return.
In case the update fails